### PR TITLE
Fixed `EVAL_PROMPT_TEMPLATE` to handle empty string or multiple match answers

### PR DESCRIPTION
**Pull request by jamesbraza**
This PR moves the evaluation prompt template to handle
- Empty string, which relates to https://github.com/Future-House/paper-qa/issues/723
- Multiple matches, which fixes https://github.com/Future-House/paper-qa/issues/709

### CODE REVIEW

**Review by sidnarayanan — COMMENTED** [review]
How do the benchmarks move with this change?

File: paperqa/prompts.py [hunk 1]
```diff
@@ -98,9 +98,12 @@
 # Prompt templates for use with LitQA
 QA_PROMPT_TEMPLATE = "Q: {question}\n\nOptions:\n{options}"
 EVAL_PROMPT_TEMPLATE = (
-    "Extract the single letter answer from the following question and answer"
```
Comment (mskarlin) [inline paperqa/prompts.py:101]: I think this is a solid iteration of the eval prompt template to address #709.

But I don't think this addresses [Empty session.answer for exhaustive search without an answer #723](https://github.com/Future-House/paper-qa/issues/723)

We still need the ability to inform the user of what happened at this step -- they just get a blank answer object currently. If the "user" is the eval script (as here) then this takes care of it.

Comment (mskarlin) [reply paperqa/prompts.py:101]: Yea, exactly -- after the runner finishes, we need some sentinel to be added. The blank isn't intuitive enough for users to understand (especially via command line or UI). Can def be another PR.

Comment (mskarlin) [reply paperqa/prompts.py:101]: In cases like this eval though -- the sentinel will create a need for another prompt, so we could make the sentinel a module level variable and check for it in the eval method.

**Comment by jamesbraza** [comment]
> How do the benchmarks move with this change?

I am going to be running this shortly. With the empty answers, previously sometimes they got marked as correct, sometimes incorrect, and sometimes unsure. I am not sure which is more frequent, but that will likely dictate the impact on benchmark performance.

Fwiw our mini test suite's benchmark has grown here.

**Comment by jamesbraza** [comment]
@sidnarayanan I will circle back with the impact of evaluation, going to merge this for now

**Comment by jamesbraza** [comment]
Okay, including all of v5.6, it seems:

- Reward: mean drops by 0.05 to 0.18, standard deviation increases by 0.06 to 0.11
- Accuracy: mean drops by about 0.2% to 5.2%, standard deviation increases 0.8% to 4%

So in general, performance drops a bit and variability increases. So the evaluation flaws in v5.0 to v5.5 were pumping up the numbers a bit

**Pull request merged**